kb/all-the-ways-to-run-a-cluster.md

```diff
@@ -0,0 +1,123 @@
+
+Running your Cluster in Docker is convenient and enables you to play with features.
+However, it does not offer the same performace in terms of processing Workflow Executions per second as Temporalite.
+Though you would only notice this if you plan to run hundreds of Workflows concurrently.
+
+The following commands start and run a Temporal Cluster in Docker using the default configuration (docker-compose.yml):
+
+```bash
+git clone https://github.com/temporalio/docker-compose.git
+cd docker-compose
+docker-compose up
+```
+
+Local [Temporal Clients](/temporal/#temporal-client) and [Workers](/workers) can connect to the Cluster running in Docker at 127.0.0.1:7233 (default connection for most SDKs) and the Temporal Web UI at 127.0.0.1:8080.
+
+To try other configurations (different dependencies and databases), or to try a custom Docker image, follow the [temporalio/docker-compose README](https://github.com/temporalio/docker-compose/blob/main/README.md).
+
+## Import the Server package
+
+The Temporal Server is a stand alone Go application which can be [imported](/references/server-options) into another project.
+
+The main reason you might want to do this is to pass in custom plugins or any other customizations through the [Server Options](/references/server-options).
+Then you can build and run a binary that contains your customizations.
+
+Doing this requires [Go v1.16+](https://github.com/temporalio/temporal/blob/master/CONTRIBUTING.md).
+
+## Temporal Server as a binary
+
+The Temporal Server can be run as a single Go binary or you can run each service within the Server separately.
+For example, if you are using Kubernetes, you would have one service per pod so they can be scaled independently in future.
+
+In Docker for example, you would run each service in its own container, using the `SERVICES` flag to specify the service:
+
+```bash
+docker run
+    # persistence/schema setup flags omitted
+    -e SERVICES=history \                      -- Spinup one or more of: history, matching, worker, frontend
+    -e LOG_LEVEL=debug,info \                           -- Logging level
+    -e DYNAMIC_CONFIG_FILE_PATH=config/foo.yaml         -- Dynamic config file to be watched
+    temporalio/server:<tag>
+```
+
+[See the Docker source file](https://github.com/temporalio/temporal/tree/master/docker) for more details.
+
+Each Temporal Server release also ships a `Server with Auto Setup` Docker image that includes [an `auto-setup.sh` script](https://github.com/temporalio/docker-builds/blob/main/docker/auto-setup.sh) we recommend using for initial schema setup of each supported database.
+You should familiarize yourself with [what auto-setup does](https://temporal.io/blog/auto-setup), as you will likely be replacing every part of the script to customize for your own infrastructure and tooling choices.
```

Comment:
```suggestion
You should familiarize yourself with [what auto-setup does](https://temporal.io/blog/auto-setup), because you will likely replace every part of the script to customize it for your own infrastructure and tooling choices.
```